frog/server/runDataflow.js

```diff
@@ -54,6 +53,8 @@ const runDataflow = (
     return;
   }
 
+  nodeTypes[type].update(nodeId, { $set: { state: 'computing' } });
```

Comment:
No, we should never have issue of concurrency as far as I understand. 

> In Meteor, your server code runs in a single thread per request, not in the asynchronous callback style typical of Node. We find the linear execution model a better fit for the typical server code in a Meteor application.

(https://gist.github.com/possibilities/3443021)